classes/services/PKPSubmissionService.php

```diff
@@ -643,4 +643,129 @@ public function compileToArrayParams($defaultParams, $params = null) {
 		return $result;
 	}
 
+	/**
+	 * Helper function to create a submission object
+	 *
+	 * @param int $contextId
+	 * @param int $submissionId
+	 * @throws Exceptions\InvalidSubmissionException
+	 *
+	 * @return Submission
+	 */
+	protected function getSubmission($contextId, $submissionId) {
```

Comment:
this is used to get a single submission based on URL parameter. But submission requests should have already been through `SubmissionAccessPolicy`. 

That being said, nothing stops another plugin to directly call the method from a request which has not been correctly authorized beforehand. I will modify the method signature and pass a submission object directly.